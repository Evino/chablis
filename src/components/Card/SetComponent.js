import React from 'react';
import Card from '.';

const baseURL = '//res.cloudinary.com/evino/image';
const imageLogo = `${baseURL}/upload/v1615413100/web/assets/Cune.svg`;
const imageCountryURL = `${baseURL}/upload/dpr_2.0,w_32,f_auto,q_auto:best/v1/web/assets/Espanha`;

const SetComponent = (bgImage, width, flag, text) => (
  <Card backgroundImage={bgImage} width={width}>
    {flag && (
      <img
        className="image-country"
        alt="Country"
        src={imageCountryURL}
      />
    )}
    <img className="image-logo" alt="Logo" src={imageLogo} />
    <span className="sr-only">{text}</span>
  </Card>
);

export default SetComponent;
