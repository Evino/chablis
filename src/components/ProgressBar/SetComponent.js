import React from 'react';
import PropTypes from 'prop-types';
import {
  ProgressBar, Bar1, Bar2, Bar3, Bar4, Bar5,
} from '.';
import BodySmall from '../Typography/BodySmall';

const SetProgressBar = ({ active, label }) => (
  <div>
    {label && <BodySmall>{label}</BodySmall>}
    <ProgressBar active={active}>
      <Bar1 />
      <Bar2 />
      <Bar3 />
      <Bar4 />
      <Bar5 />
    </ProgressBar>
  </div>
);

SetProgressBar.propTypes = {
  active: PropTypes.string,
  label: PropTypes.string,
};

export default SetProgressBar;
