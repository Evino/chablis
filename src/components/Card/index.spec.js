import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import SetComponent from './SetComponent';

describe('Card', () => {
  it('Should render the default card', () => {
    const element = SetComponent('bg-image', '337px', true, 'Default card');
    const tree = renderer.create(element).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Should render the card without flag', () => {
    const element = SetComponent('bg-image', '337px', false, 'Card without flag');
    const component = mount(element);
    expect(component.find('.image-country')).toHaveLength(0);
  });

  it('Should render the component without background image', () => {
    const element = SetComponent('', '337px', true, 'Card without background image');
    const component = mount(element);
    expect(component.props().backgroundImage).toEqual('');
  });

  it('Should render a fluid component', () => {
    const element = SetComponent('', '', true, 'Card fluid');
    const component = mount(element);
    expect(component.props().width).toEqual('');
  });
});
