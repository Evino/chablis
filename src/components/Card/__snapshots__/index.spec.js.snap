// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Should render the default card 1`] = `
<div
  className="sc-bdVaJa bsZmUM"
  width="337px"
>
  <img
    alt="Country"
    className="image-country"
    src="//res.cloudinary.com/evino/image/upload/dpr_2.0,w_32,f_auto,q_auto:best/v1/web/assets/Espanha"
  />
  <img
    alt="Logo"
    className="image-logo"
    src="//res.cloudinary.com/evino/image/upload/v1615413100/web/assets/Cune.svg"
  />
  <span
    className="sr-only"
  >
    Default card
  </span>
</div>
`;
